feat(socket): 소켓 의존성 및 `useSocket` 추가

hakilee opened the pull request:
## 🔗 반영 브랜치
- closed #114
- `feature/web-socket → dev`

## 📝 작업 내용

소켓을 조금 더 개발 친화적으로 사용할 수 있도록 `useSocket` 훅을 추가했습니다.
훅 생성 시에 `createSoekct` 함수를 호출하게 되는데, 이를 싱글톤으로 구현해 중복 연결을 방지하고, 리소스 낭비를 줄였습니다.

hakilee committed:
feat: add socket connectors

hakilee committed:
chore: add socket base url

hakilee committed:
Merge remote-tracking branch 'origin/dev' into feature/web-socket

# Conflicts:
#	src/App.tsx

hakilee committed:
feat: 소켓 연결 추가

hakilee committed:
feat: 소켓 연결 및 관리 로직을 useSocket 훅으로 분리

FriendPage에서 직접 소켓을 생성하고 관리하는 대신, `useSocket` 훅을 사용하여 소켓 연결을 관리하도록 변경했습니다.

- 소켓 생성 및 연결 로직을 재사용 가능한 훅으로 분리했습니다.
- 컴포넌트에서 소켓 로직을 분리하여 코드 가독성과 유지보수성을 향상시켰습니다.
- 소켓 연결 상태를 UI에 표시하도록 변경했습니다.

hakilee committed:
Merge remote-tracking branch 'origin/dev' into feature/web-socket

# Conflicts:
#	src/App.tsx

hakilee committed:
refactor: 소켓 생성 로직 및 useSocket훅 수정

`getSocket` 함수명을 `createSocket`으로 변경하고, 소켓 생성 로직을 useSocket 훅 내부에서 직접 호출하도록 수정했습니다. 이를 통해 코드의 가독성을 향상시키고 불필요한 의존성을 제거했습니다. FriendPage 컴포넌트에서 import 경로를 수정했습니다.

hakilee committed:
chore: 종속성 업데이트

@tanstack/react-query, jotai, react-router-dom 등 주요 라이브러리 버전 업데이트.
package.json 및 pnpm-lock.yaml 파일 수정.

hakilee commented on vite.config.ts:
```diff
@@ -28,6 +28,7 @@ export default defineConfig(({ mode }) => {
         '/api': {
           target: API_URL,
           changeOrigin: true,
+          secure: false,
```
서버에 인증서 추가 시 제거해야 해요.

hakilee commented on src/pages/FriendPage/index.tsx:
```diff
@@ -0,0 +1,18 @@
+    withToken: true,
+  });
+
+  socket.connect();
```
테스트 용도로 작성된 라인이에요.

yeni023 reviewed (dismissed):
제가 전체 로직을 아래처럼 이해했는데 혹시 맞게 이해한 걸까요?

1. `useSocket` 훅이 호출되면 내부에서 `createSocket`을 통해 소켓 인스턴스를 생성하거나 이미 있는 경우 캐싱된 인스턴스를 재사용
2. `createSocket`은 `path`, `handshake`, `token` 등을 기준으로 URL을 조합한 key로 소켓을 관리해서 중복 연결을 방지

hakilee committed:
feat: 친구 페이지 추가 및 소켓 연결 상태 표시 기능 구현

친구 페이지를 추가하고, 해당 페이지에서 소켓 연결 상태를 실시간으로 표시하는 기능을 구현했습니다.

- App.tsx: 친구 페이지 라우팅 추가
- src/pages/FriendPage/index.tsx: 친구 페이지 컴포넌트 생성 및 소켓 연결/해제 로직 추가, 연결 상태 표시
- src/api/socket/useSocket.ts: 소켓 연결 상태를 관리하고, 연결/해제 시 상태 업데이트 로직 추가

yeni023 reviewed (approved):
확인했습니다!
